wordpress-mobile/WordPress-Android
# [Jetpack focus] Implement shared login credentials analytics

Fixes #17126

To test:

1 - Install WordPress and login;
2 - [Set the flag `isFeatureFlagEnabled`](https://github.com/wordpress-mobile/WordPress-Android/pull/17059/files#diff-90a223fc6cd4e93f8b73c9f73663fcd96d804b4acbc420af6be322bac2388948R32) to `true` (this cannot be done using the `Debug settings` screen because that is only available when the user is already logged in);
2 - Install Jetpack using the same variant (e.g. if installed WordPress version was `jalapenoDebug`, install Jetpack `jalapenoDebug`);
3 - Jetpack should login automatically after launching and the events `"shared_login_start"` and `"shared_login_success"` should be triggered;
4 - Close the app and open it again: `"shared_login_start"` and  `"shared_login_success"` should NOT be triggered again
5 - Clear storage on Jetpack app, [set `accessTokenResultCursor`](https://github.com/wordpress-mobile/WordPress-Android/compare/issue/17126-shared-login-credentials-analytics?expand=1#diff-90a223fc6cd4e93f8b73c9f73663fcd96d804b4acbc420af6be322bac2388948R40) to `null`, build and install Jetpack again;
6 - Jetpack should NOT login automatically after launching and the events `"shared_login_start"` and `"shared_login_failed"` should be triggered;
7 - Close the app and open it again: `"shared_login_start"` and  `"shared_login_failed"` should NOT be triggered again;
8 - Clear storage on Jetpack app, [set `accessToken`](https://github.com/wordpress-mobile/WordPress-Android/compare/issue/17126-shared-login-credentials-analytics?expand=1#diff-90a223fc6cd4e93f8b73c9f73663fcd96d804b4acbc420af6be322bac2388948R42) to empty (`""`), build and install Jetpack again;
9 - Jetpack should NOT login automatically after launching and the events `"shared_login_start"` and `"shared_login_failed"` should be triggered;
10 - Close the app and open it again: `"shared_login_start"` and  `"shared_login_failed"` should NOT be triggered again;

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and unit testing

3. What automated tests I added (or what prevented me from doing so)
Updated `SharedLoginResolverTest.kt`

PR submission checklist:
